---
title: "Protist communities along a surface-groundwater gradient in the Danube wetlands of Vienna" 
author:  "Angela Cukusic"
output:
  html_document:
    toc: true               # Table of contents
    toc_float: true         # Keep the table of contents floating on the page
    code_download: true     # Enable code download button
    code_copy: true         # Enable code copy button
    number_sections: true   # Numbering the sections
    theme: cerulean           # Choose a Bootstrap theme (check available themes)
    highlight: tango        # Syntax highlighting style 
---

```{r setup, include=FALSE}
# For width of code chunks and scroll bar 
options(width=250)

knitr::opts_chunk$set(eval = TRUE, 
                      echo = TRUE, 
                      cache = TRUE,
                      include = TRUE,
                      warning = FALSE,
                      collapse = FALSE,
                      message = FALSE,
                      engine = "R", 
                      # Chunks will always have R code, unless noted
                      error = TRUE)

```

# load packages
```{r}
library(tidyverse)
library(stringr)
library(decontam)
library(readxl)
library(vegan)
library(patchwork)
library(ggpmisc)
library(MASS)
library(microViz)
library(microbiome)
library(car)
library(corrplot)
library(magrittr)
library(compositions)
library(microeco)
library(cowplot)



pal <- c("gray90", "gray20")
```


# load data
```{r}
# asv table
asv_table_SSU <-  read.csv("~/faks/poster_fungi/poster_fungi_jmf_2023-05-16/JMF-2301-09__all__rRNA_SSU_Euk_D__JMFR_MSHI_KP237/DADA2_counts_as_matrix.tsv", row.names=1,  check.names = FALSE, sep = "\t")
names(asv_table_SSU)<- sub("C","", names(asv_table_SSU))
# the first 78 samples are rna, so we take the second half
asv_table <- asv_table_SSU[, -c(1:78)]
asv_table_rna <- asv_table_SSU[, c(1:78)] 

# tax table
tax_table_SSU_RP2 <- read.delim("~/faks/poster_fungi/poster_fungi_jmf_2023-05-16/JMF-2301-09__all__rRNA_SSU_Euk_D__JMFR_MSHI_KP237/DADA2_ASVs.rRNA_SSU.PR2_reference.DADA2_classified.tsv", row.names=1)

#env table
env_data <- read_excel("~/faks/christina/data/data_angela/env_combined.xlsx") 
```

# clean up

## remove the blanks
```{r}
jmf_blank_info <- read_excel("~/faks/christina/data/data_angela/jmf_2301_09_esw_d15.xlsx") 


jmf_blank_info <- jmf_blank_info %>% 
  dplyr::select("JMF sample ID", "Sample description", "Date") %>% 
  dplyr::rename("Sample_desc" = "Sample description",
          "JMF.ID" = "JMF sample ID") %>% 
# add blank or sample column  
  mutate(blank_sample = case_when(startsWith(Sample_desc, "Blank") ~ "blank",
                                             TRUE ~ "sample")) %>% 
  dplyr::filter(blank_sample == "sample")
 
 
# remove blanks from the asv table
asv_table <- asv_table[, colnames(asv_table) %in% jmf_blank_info$JMF.ID]
asv_table_rna <- asv_table_rna[, colnames(asv_table_rna) %in% jmf_blank_info$JMF.ID] 
```




## connect sequincing id to sample_id
```{r}

# find the unique sample name first based on well id and sampling date
jmf_blank_info <- jmf_blank_info  %>% 
  mutate (well_id = case_when(startsWith(Sample_desc, "ESW") ~ "ESW_", 
                              startsWith(Sample_desc, "D05") ~ "D05p_", 
                              startsWith(Sample_desc, "D10") ~ "D10p_", 
                              startsWith(Sample_desc, "D15") ~ "D15p_"   )) %>% 
   mutate(sample_id = paste(well_id, Date)) %>% 
   dplyr::filter(well_id %in% c("ESW_", "D15p_"))
jmf_blank_info$sample_id <- gsub(" ", "", jmf_blank_info$sample_id)
jmf_blank_info$sample_id <- gsub("-", "", jmf_blank_info$sample_id)

# clean the Unbekannt one (JMF-2301-09-0109)that we do not know the id of
jmf_blank_info <- jmf_blank_info %>% dplyr::filter(Sample_desc != "Unbekannt")


env_lobau_final <- inner_join(jmf_blank_info, env_data, by = "sample_id")
rownames(env_lobau_final) <- env_lobau_final$JMF.ID

# rename the asv_table also
asv_table %>% 
  rownames_to_column("taxa") %>% 
  pivot_longer(cols = - taxa, names_to = "JMF.ID", values_to = "counts" ) %>% 
  left_join(., jmf_blank_info[,c("sample_id", "JMF.ID")], by = "JMF.ID") %>% 
  dplyr::select(!"JMF.ID") %>% 
# if there is replicates for the same sample_id add it up  
  group_by(taxa, sample_id) %>%
  summarise(counts = sum(counts)) %>%
  ungroup() %>% 
  pivot_wider(names_from = "sample_id", values_from = "counts") %>%
  column_to_rownames(var="taxa") -> asv_table


# the last column is NA -  the Unbekannt one?
# cannot be sure so just remove it
asv_table <- asv_table[, 1:46]


asv_table_rna %>% 
  rownames_to_column("taxa") %>% 
  pivot_longer(cols = - taxa, names_to = "JMF.ID", values_to = "counts" ) %>% 
  left_join(., jmf_blank_info[,c("sample_id", "JMF.ID")], by = "JMF.ID") %>% 
  dplyr::select(!"JMF.ID") %>% 
# if there is replicates for the same sample_id add it up  
  group_by(taxa, sample_id) %>%
  summarise(counts = sum(counts)) %>%
  ungroup() %>% 
  pivot_wider(names_from = "sample_id", values_from = "counts") %>%
  column_to_rownames(var="taxa") -> asv_table_rna


```


## clean up env data
```{r}
# since we removed the replicates from asv_table remove them from env_table

env_lobau_final %>%  dplyr::filter(! JMF.ID %in% c("JMF-2301-09-0152", 
                                                   "JMF-2301-09-0147", 
                                                   "JMF-2301-09-0133",
                                                   "JMF-2301-09-0082", 
                                                   "JMF-2301-09-0083",
                                                   "JMF-2301-09-0084", 
                                                   "JMF-2301-09-0102", 
                                                   "JMF-2301-09-0106", 
                                                   "JMF-2301-09-0104", 
                                                   "JMF-2301-09-0139",
                                                   "JMF-2301-09-0100", 
                                                   "JMF-2301-09-0092", 
                                                   "JMF-2301-09-0035", 
                                                   # rna replicates
                                                        "JMF-2301-09-0016",
                                                        "JMF-2301-09-0026", 
                                                        "JMF-2301-09-0028", 
                                                        "JMF-2301-09-0027", 
                                                        "JMF-2301-09-0004", 
                                                        "JMF-2301-09-0029",  
                                                        "JMF-2301-09-0055", 
                                                        "JMF-2301-09-0030", 
                                                        "JMF-2301-09-0032",
                                                        "JMF-2301-09-0033") ) -> env_lobau_final

# since i know which ones are replicates it is easiest for me to just remove those jmfs




env_lobau_final <- env_lobau_final %>%
# add dna vs rna value  
  mutate(active_comm = ifelse(row_number() <= 49, "rna", "dna")) %>% 
# add season category  
mutate(season = case_when ( 
     grepl('2021-05|2021-04|2022-04|2021-03|2022-03|2022-03|2022-05', Date) ~ "spring",
     grepl('2020-07|2022-08|2021-06|2022-06|2021-07|2022-08', Date)  ~ "summer",
     grepl('2020-09|2020-10|2022-10|2021-10|2022-11|2020-11|2021-11', Date) ~ "fall",
     grepl('2020-12|2021-12|2021-01|2022-01|2021-02|2022-02', Date) ~ "winter"   ))
```

We are left with 46 samples for dna and 49 for rna.


## clean up tax data
For now just remove the Metazoa, Fungi and unclassified divisions.
```{r}
tax_table <- tax_table_SSU_RP2 %>% 
  dplyr::filter(!Supergroup == "") %>% 
  dplyr::filter(!Division %in% c("Fungi", "Metazoa", "") )

# remove the taxa also from asv_table
asv_table <- asv_table[which(rownames(asv_table) %in% rownames(tax_table)) ,]
asv_table_rna <- asv_table_rna[which(rownames(asv_table_rna) %in% rownames(tax_table)) ,]

```




# Read numbers
```{r}

# dna
read_data <- asv_table %>% 
colSums() %>% # how many reads per sample (find sums per each column)
as.data.frame() %>% 
rownames_to_column("sample_id") %>% 
dplyr::rename("reads" = ".") %>% 
mutate (well_id = case_when(startsWith(sample_id, "ESW") ~ "ESW",  
                              startsWith(sample_id, "D15") ~ "D15"   ))

summary(read_data)
sd(read_data$reads)

boxplot(read_data$reads ~ read_data$well_id)


read_data %>% 
  dplyr::filter(reads < 60000) %>% 
  group_by(well_id) %>% 
  summarise(sd = sd(reads), mean = mean(reads) )


# rna
read_data <- asv_table_rna %>% 
colSums() %>% # how many reads per sample (find sums per each column)
as.data.frame() %>% 
rownames_to_column("sample_id") %>% 
dplyr::rename("reads" = ".") %>% 
mutate (well_id = case_when(startsWith(sample_id, "ESW") ~ "ESW",  
                              startsWith(sample_id, "D15") ~ "D15"   ))

summary(read_data)
sd(read_data$reads)

boxplot(read_data$reads ~ read_data$well_id)


read_data %>% 
  dplyr::filter(reads < 60000) %>% 
  group_by(well_id) %>% 
  summarise(sd = sd(reads), mean = mean(reads) )

```

# Count numbers
Search for the count numbers - considering the elevated numbers of contaminated aquifers in comparison to pristine aquifers, it could be hypothesized there is a bigger count number in the surface water?
```{r}

# sample sizes
table(env_lobau_final$season, env_lobau_final$well_id.y)
# fairly similar sample sizes



asv_table %>% 
  rownames_to_column("taxa") %>% 
  pivot_longer(cols = -taxa, values_to = "counts", names_to = "sample_id") %>% 
  left_join(., env_lobau_final[, c("sample_id", "well_id.y", "season")], by = "sample_id") %>% 
  
# do not count by individual taxa, look at per sample count  
  group_by(sample_id) %>% 
  mutate(counts = sum(counts)) %>% 
  ungroup() %>% 
  
  dplyr::filter(counts <60000) %>% 
  
  ggplot(aes(x= well_id.y, y=counts, fill = well_id.y)) +
  geom_boxplot(alpha = 0.7) + 
  scale_y_log10(labels =  function(x) format(x, scientific = TRUE)) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(), 
        legend.background = element_blank(),
        legend.box.background = element_rect(colour = "gray"),
        legend.position =c(0.1, 0.9),
        axis.text.x = element_text(face = "bold", size = 12), 
        axis.text.y = element_text(face = "bold", size = 11), 
        axis.title = element_text(face = "bold", size = 14),
        legend.text = element_text(size = 11) ) +
  scale_fill_manual(values = pal) +
  labs(x = "Well ID", y = "Total 18S counts of sample",
       fill = "Well ID")   -> p_counts2


```


```{r}
asv_table %>% 
  rownames_to_column("taxa") %>% 
  pivot_longer(cols = -taxa, 
               values_to = "counts", 
               names_to = "sample_id") %>% 
  left_join(., env_lobau_final[, c("sample_id", "well_id.y", "season")], 
            by = "sample_id") %>% 
  
# do not count by individual taxa, look at per sample count  
  group_by(sample_id, well_id.y) %>% 
  summarise(counts = sum(counts)) %>% 
  ungroup() -> data3



data3[data3$well_id.y == "ESW","counts"] %>% as.data.frame -> counts_esw
data3[data3$well_id.y == "D15","counts"] %>% as.data.frame -> counts_D15

# Shapiro-Wilk test
shapiro.test(counts_esw$counts)
shapiro.test(counts_D15$counts)

# Levene's test
leveneTest(data3$counts ~ well_id.y, data=data3)

# not normal - use non parametric

wilcox.test(counts_esw$counts, counts_D15$counts)
  
```


## by season
```{r}

asv_table %>% 
  rownames_to_column("taxa") %>% 
  pivot_longer(cols = -taxa, values_to = "counts", names_to = "sample_id") %>% 
  left_join(., env_lobau_final[, c("sample_id", "well_id.y", "season")], by = "sample_id") %>% 
  
# do not count by individual taxa, look at per sample count  
  group_by(sample_id) %>% 
  mutate(counts = sum(counts)) %>% 
  ungroup() %>% 
  
  ggplot(aes(x= season, y=counts, fill = well_id.y)) +
  geom_boxplot(alpha = 0.7) + 
  scale_y_log10(labels =  function(x) format(x, scientific = TRUE)) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(), 
        legend.background = element_blank(),
        legend.box.background = element_rect(colour = "gray"),
        legend.position =c(0.1, 0.9),
        axis.text.x = element_text(face = "bold", size = 12), 
        axis.text.y = element_text(face = "bold", size = 11), 
        axis.title = element_text(face = "bold", size = 14),
        legend.text = element_text(size = 11) ) +
  scale_fill_manual(values = pal) +
  labs(x = "Season", y = "Total 18S counts of sample",
       fill = "Well ID") +
  scale_x_discrete(limits = c("fall", "winter", "spring", "summer"))   -> p_counts
p_counts
```



# Rarefaction
```{r}
# dna
rrarefy(t(asv_table), sample = 4000) %>% t() %>% as.data.frame() -> asv_table_r

asv_table_r %>% colSums() %>%  plot()

# there is 6 sample that have less than 4000 counts
# remove them from asv_table
asv_table_r[,colSums(asv_table_r) == 4000] -> asv_table_r



# rna 
rrarefy(t(asv_table_rna), sample = 1080) %>% t() %>% as.data.frame() -> asv_table_rna_r

asv_table_rna_r %>% colSums() %>%  plot()

# there is 9 sample that have less than 2000 counts
# remove them from asv_table
asv_table_rna_r[,colSums(asv_table_rna_r) == 1080] -> asv_table_rna_r





# remove them from env data
# dna
env_lobau_final %>% dplyr::filter(active_comm == "dna") -> env_lobau_final_r
env_lobau_final_r[env_lobau_final_r$sample_id %in%  
                    c(colnames(asv_table_r)),] -> env_lobau_final_r


# rn
env_lobau_final %>% dplyr::filter(active_comm == "rna") -> env_lobau_final_rna_r
env_lobau_final_rna_r[env_lobau_final_rna_r$sample_id %in%  c(colnames(asv_table_rna_r)),] -> env_lobau_final_rna_r
```



# ordering
```{r}
env_lobau_final_r[order(env_lobau_final_r$sample_id), ] -> env_lobau_final_r
identical(env_lobau_final_r$sample_id, colnames(asv_table_r))


env_lobau_final_rna_r[order(env_lobau_final_rna_r$sample_id), ] -> env_lobau_final_rna_r
identical(env_lobau_final_rna_r$sample_id, colnames(asv_table_rna_r))


```


# Diversity
```{r}

shannondiv <- vegan::diversity(t(asv_table_r))

as.data.frame(shannondiv) %>% 
  rownames_to_column("sample_id") %>% 
  left_join(., env_lobau_final_r, by = "sample_id") %>% 
# plot
  ggplot(aes(x= season, y=shannondiv, fill = well_id.y)) +
  geom_boxplot(alpha = 0.7) + 
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(), 
        legend.background = element_blank(),
        legend.box.background = element_rect(colour = "gray"),
        legend.position = c(0.1, 0.1),
        axis.text.x = element_text(face = "bold", size = 12), 
        axis.text.y = element_text(face = "bold", size = 11), 
        axis.title = element_text(face = "bold", size = 14)) +
  scale_fill_manual(values = pal) +
  labs(x = "Season", y = NULL, fill = "Well ID")  +
  scale_x_discrete(limits = c("fall", "winter", "spring", "summer")) -> p_div

p_div


```

```{r}

as.data.frame(shannondiv) %>% 
  rownames_to_column("sample_id") %>% 
  left_join(., env_lobau_final_r, by = "sample_id") -> data3



# fall
wilcox.test(data3 %>% dplyr::filter(season == "fall" & well_id.y == "D15") %>% .$shannondiv, 
           data3 %>% dplyr::filter(season == "fall" & well_id.y == "ESW")  %>% .$shannondiv )

# spring
wilcox.test(data3 %>% dplyr::filter(season == "spring" & well_id.y == "D15") %>% .$shannondiv, 
           data3 %>% dplyr::filter(season == "spring" & well_id.y == "ESW")  %>% .$shannondiv )

# summer
wilcox.test(data3 %>% dplyr::filter(season == "summer" & well_id.y == "D15") %>% .$shannondiv, 
           data3 %>% dplyr::filter(season == "summer" & well_id.y == "ESW")  %>% .$shannondiv )


```


```{r}

as.data.frame(shannondiv) %>% 
  rownames_to_column("sample_id") %>% 
  left_join(., env_lobau_final_r, by = "sample_id") %>% 
# plot
  ggplot(aes(x= well_id.y, y=shannondiv, fill = well_id.y)) +
  geom_boxplot(alpha = 0.7) + 
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(), 
        legend.background = element_blank(),
        legend.box.background = element_rect(colour = "gray"),
        legend.position ="none",
        axis.text.x = element_text(face = "bold", size = 12), 
        axis.text.y = element_text(face = "bold", size = 11), 
        axis.title = element_text(face = "bold", size = 14)) +
  scale_fill_manual(values = pal) +
  labs(x = "Well ID", y = "Shannon's diversity index")  -> p_div2

p_div2

```

```{r}

data3[data3$well_id.y == "ESW","shannondiv"] %>% as.data.frame -> ens_esw
data3[data3$well_id.y == "D15","shannondiv"] %>% as.data.frame -> ens_D15

# Shapiro-Wilk test
shapiro.test(ens_esw$.) # normal
shapiro.test(ens_D15$.) # not normal

# Levene's test
leveneTest(data3$shannondiv ~ well_id.y, data=data3)

# not normal - use non parametric
wilcox.test(ens_esw$., ens_D15$.)
```


```{r}

p_div2 + p_div  + plot_layout(widths = c(1,2))
```

# nmds
```{r}

# do an ordination with a hellinger transformation 
sp.dist <-  vegdist(decostand( t(asv_table_r),
                               method = "hellinger") )



nmds <- metaMDS(sp.dist, k = 2, autotransform = FALSE)
plot(nmds)


as.data.frame(nmds$points) %>% 
  cbind(., env_lobau_final_r) %>% 
  
  ggplot(aes(x=MDS1,y=MDS2) ) +
  geom_point(aes( fill= well_id.y, shape = season), 
             size = 4, alpha = 0.6, stroke = 0.7) +  
# coloring  
  #scale_color_manual(values=pal) +
  #scale_fill_manual(values=pal) +
# add stress value
  annotate(geom="text",x=0.7, y=1.2, 
           label="stress =", color="black") +   
  annotate(geom="text", x=1.1, y=1.2, 
           label=round(nmds$stress,4), color="black") +
# ellipse  
  stat_ellipse(aes(colour = well_id.y), linewidth = 0.9) + 
  theme_bw() +
# visual parameters 
  theme(panel.grid = element_blank(),
        legend.background = element_blank(),
        legend.box.background = element_rect(colour = "gray")) +
  labs(color = "Well ID", shape = "Season") +
  guides(fill = "none") + 
  scale_shape_manual(values = c(21, 22, 23, 24)) +
  scale_color_manual(values = pal) +
  scale_fill_manual(values = pal) 

```



```{r}

# well id
# first test assumption: equal dispersion
disp_aq <- betadisper( 
  vegdist(t(asv_table_r),
          method = "bray"), 
          env_lobau_final_r$well_id.y ) 

#p>0.05 is what we need - homogenous variances
anova(disp_aq) #no sign. differences in dispersion


# now test actual ordination
permanova_aq <- adonis2(t(asv_table_r) ~ 
          env_lobau_final_r$well_id.y , 
                         method = "bray" , 
                        permutations = 999)
permanova_aq # not sign


# season
# first test assumption: equal dispersion
disp_aq <- betadisper( 
  vegdist(t(asv_table_r),
          method = "bray"), 
          env_lobau_final_r$season ) 

#p>0.05 is what we need - homogenous variances
anova(disp_aq) #no sign. differences in dispersion


# now test actual ordination
permanova_aq <- adonis2(t(asv_table_r) ~ 
          env_lobau_final_r$season , 
                         method = "bray" , 
                        permutations = 999)
permanova_aq # not sign





adonis2(t(asv_table_r) ~ 
          env_lobau_final_r$season *  env_lobau_final_r$well_id.y, 
                         method = "bray" , 
                        permutations = 999)

```



# env variables


```{r}
my_labeller2 <- as_labeller(c(
  ec_uS_cm = "EC ~ (µS/cm)",
  pH = "pH",
  oxygen_mg_L="O[2] ~ (mg/L)",
  doc_mg_L = "DOC ~ (mg/L)",
  dic_mg_L = "DIC ~ (mg/L)", 
  co2_uM = "CO[2] ~ (uM)"),
  default = label_parsed)

my_labeller1 <- as_labeller(c(
 atp_internal_pM = "ATP ~  (pM/L)",
  cells_per_mL = "TCC ~ (cells/mL)"),
  default = label_parsed)


        

```


```{r}
env_lobau_final_r %>% 
  dplyr::select("well_id.y", "pH", "ec_uS_cm", "oxygen_mg_L", "doc_mg_L", "dic_mg_L", 
                 "co2_uM") %>% 
  dplyr::filter(doc_mg_L < 30) %>% 
dplyr::filter(co2_uM < 1000) %>% 
pivot_longer(cols = -well_id.y, values_to = "values", names_to = "names") %>% 
  ggplot(aes(x= well_id.y, y= values, fill = well_id.y)) +
  geom_boxplot(alpha = 0.7) +
  facet_wrap(~names, scales = "free_y", ncol = 2,  
             strip.position = "left", 
             labeller = my_labeller2) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        legend.position = "none",
         axis.text.x = element_text(face = "bold", size = 12), 
        axis.text.y = element_text(face = "bold", size = 11), 
        axis.title = element_text(face = "bold", size = 14),
        plot.margin = unit(c(0, 0.5, 0, 0), "null"), 
        strip.text = element_text(size = 12)) +
  scale_fill_manual(values = pal) + 
  labs(y = NULL, x = "Well ID") -> plot_rest


env_lobau_final_r %>% 
  dplyr::select("well_id.y", "atp_internal_pM",
                "cells_per_mL") %>% 
pivot_longer(cols = -well_id.y, values_to = "values", names_to = "names") %>% 
  ggplot(aes(x= well_id.y, y= values, fill = well_id.y)) +
  geom_boxplot(alpha = 0.7) +
  scale_y_log10() +
   facet_wrap(~names, scales = "free_y", ncol = 2,  
              strip.position = "left", 
             labeller = my_labeller1) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        legend.position = "none",
         axis.text.x = element_blank(), 
        axis.text.y = element_text(face = "bold", size = 11), 
        axis.title = element_text(face = "bold", size = 14),
        axis.ticks.x = element_blank(),
        strip.text = element_text(size = 12)) +
  scale_fill_manual(values = pal) + 
  labs(y = NULL, x = NULL) -> plot_atp

plot_atp + plot_rest + plot_layout(heights = c(1,3))

```
# bc vs env
```{r}
aa <- asv_table_r %>%  
      t() %>% 
      vegdist(., method = "bray") %>%
      as.vector()

env_lobau_final_r %>% 
  as.data.frame() %>% 
  column_to_rownames("sample_id") %>% 
  dplyr::select("pH", "temp_C", "ec_uS_cm", "oxygen_mg_L",
                "doc_mg_L", "dic_mg_L", "soluble_P_ug_L", 
                "co2_uM", "cells_per_mL", "atp_internal_pM")%>% 
  scale() %>% as.data.frame() -> env_euc

env_euc %>%
  rownames_to_column("asv") %>% 
  pivot_longer(cols = -asv, values_to = "values",
               names_to = "variables") %>% 
  group_by(variables) %>% 
  summarise(distmatrix=list(dist(values) )) -> dfg


for (i in 1:10) {
  # Set the y values to the current column
  y_values <- dfg[i, "distmatrix"] %>%  unlist() %>% as.vector() 
  
  # Create a plot
  plot(aa, y_values, type = "p",  pch = 1,
       col = "blue", xlab = "X-axis", ylab = "Y-axis", 
       main = dfg[i,"variables"]) +
    abline(lm(y_values ~ aa ))
  
  # Extract the R
  adj_r_squared <- summary(lm(y_values ~ aa ))$adj.r.squared
  
  # Add the slope coefficient as text to the plot
  text(max(y_values - 1), pos = 2,
       paste("R2 =", round(adj_r_squared, 2)), 
       col = "red")

}  
```



Relationship of oxygen and composition variability seems most promising, but there is also relationship with atp, cells, and pH.


```{r}
gg <- dist(env_lobau_final_r$oxygen_mg_L %>% na.omit(), 
           method = "euclidean" )  %>%
      as.vector()

aa <- asv_table_r[,-33] %>% # this is the sample without oxygen  
      t() %>% 
      vegdist(., method = "bray") %>%
      as.vector() 

data.frame(aa,  gg) %>% 
  ggplot(aes(x=aa, y=gg)) + 
  geom_point(color = "black", shape = 21, size = 4, 
             fill = "gray", alpha = 0.4) +
  geom_smooth(method = "lm", fullrange = FALSE) +
  # add correlation info  
  ggpubr::stat_cor(method = "spearman",  
                   label.y.npc="top", 
                   label.x.npc = "left",
                   color="black", size=4) +
  theme_bw() +
  labs(x= "Bray-Curtis dissimilarity", y = "Oxygen gradient") +
  coord_cartesian(ylim = c(0, 13))
```


# composition

## by sample site
```{r}

library(phyloseq)
library(microshades)

as.data.frame(env_lobau_final_r) %>% column_to_rownames("sample_id") -> env.data

ps_dna <- phyloseq(otu_table(as.matrix(asv_table_r), taxa_are_rows=TRUE), 
                   tax_table(as.matrix(tax_table) ) , 
                   sample_data(env.data ) )



 mdf_prep <- ps_dna %>%
         tax_glom("Class") %>%  
         merge_samples( group = "well_id.y") %>%
         phyloseq::transform_sample_counts(function(x) { x/sum(x) }) %>%
         psmelt() %>%
         filter(Abundance > 0) %>% 
         dplyr::select("Abundance", "Sample", "Division", "Class") %>%
         mutate(Division = as.factor(Division),
                Class = as.factor(Class))
 
# find the 4 most abundant division 
 mdf_prep %>% group_by(Division) %>% 
   summarise(Abundance = sum (Abundance)) %>% 
   slice_max(order_by =Abundance, n= 4) %>% 
   as.data.frame() %>% 
   dplyr::select("Division") %>% unique()
 # c( Ochrophyta, Ciliophora, Discoba, Conosa, Dinoflagellata,	Cryptophyta, Cercozoa )
 
 
 

color_objs_GP <- create_color_dfs(mdf_prep,selected_groups = c("Ochrophyta", 
                                                               "Ciliophora", 
                                                               "Discoba",
                                                               "Conosa", 
                                                               #"Dinoflagellata",
                                                               #"Cryptophyta",
                                                               "Cercozoa"
                                                               ) , cvd = TRUE, 
                                  group_level = "Division", subgroup_level = "Class",
                                  top_n_subgroups = 4)

# Extract
mdf_GP <- color_objs_GP$mdf
cdf_GP <- color_objs_GP$cdf


plot <- plot_microshades(mdf_GP, cdf_GP) +
  scale_y_continuous(labels = scales::percent, 
                     expand = expansion(0)) +
  theme_bw() +
  theme(legend.position = "none",
        panel.grid = element_blank(),
         axis.text.x = element_text(face = "bold", size = 12), 
        axis.text.y = element_text(face = "bold", size = 11), 
        axis.title = element_text(face = "bold", size = 14)) 
plot


GP_legend <- custom_legend(mdf_GP, 
                          cdf_GP, 
                          group_level = "Division", 
                          subgroup_level = "Class",
                          legend_key_size = 1,
                          legend_text_size = 12)
cowplot::plot_grid(plot, GP_legend,  rel_widths = c(1, .5))


```


## by season
```{r}
mdf_prep_s <- ps_dna %>% tax_glom("Class") %>% 
  psmelt() %>% 
  group_by(season, well_id.y, Class) %>% 
  mutate(Abundance = mean (Abundance)) %>% 
  ungroup()  %>% 
  dplyr::group_by(season, well_id.y) %>%
  mutate (rel_abund = Abundance / sum(Abundance)) %>%
  ungroup() %>% 
  dplyr::select( "rel_abund", "well_id.y", "Division", "Class" , "season") %>% 
  dplyr::rename("Sample" = "well_id.y",
                "Abundance" = "rel_abund") %>% 
  as.data.frame()



# find the 4 most abundant division 
 mdf_prep_s %>% group_by(Division) %>% 
   summarise(Abundance = sum (Abundance)) %>% 
   slice_max(order_by = Abundance, n = 4) %>% 
   as.data.frame() %>% 
   dplyr::select("Division") %>% unique()
 # c( Ochrophyta, Ciliophora, Discoba, Conosa, Dinoflagellata,	Cryptophyta, Cercozoa )
 
 
 

color_objs_GP <- create_color_dfs(mdf_prep_s, 
                                  selected_groups = c("Ochrophyta", 
                                                      "Ciliophora", 
                                                      "Discoba",
                                                      "Conosa", 
                                                      #"Dinoflagellata",
                                                      #"Cryptophyta",
                                                      "Cercozoa" ), 
                                  cvd = TRUE, 
                                  group_level = "Division", 
                                  subgroup_level = "Class",
                                  top_n_subgroups = 4)

# Extract
mdf_GP <- color_objs_GP$mdf 
# dataset with      Abundance Sample  Division   
cdf_GP <- color_objs_GP$cdf



plot <- plot_microshades(mdf_GP, cdf_GP) +
  scale_y_continuous(labels = scales::percent, 
                     expand = expansion(0)) +
  theme_bw() +
  facet_grid(~season) +
  theme(legend.position = "none",
        panel.grid = element_blank(),
         axis.text.x = element_text(face = "bold", size = 12), 
        axis.text.y = element_text(face = "bold", size = 11), 
        axis.title = element_text(face = "bold", size = 14)) 


GP_legend <- custom_legend(mdf_GP, 
                          cdf_GP, 
                          group_level = "Division", 
                          subgroup_level = "Class",
                          legend_key_size = 1,
                          legend_text_size = 12)
cowplot::plot_grid(plot, GP_legend,  rel_widths = c(1, .5))
```

## by season (only Conosa and Discoba)
```{r}
mdf_prep_s %>% 
  dplyr::filter(Division %in% c("Conosa", "Discoba")) %>% 
  dplyr::filter(Sample == "D15") %>% 
  group_by(Division, season) %>% 
  summarise(Abundance = sum(Abundance)) %>% 
  ungroup() %>% 
  ggplot(aes(x=season, y=Abundance, group=Division, fill  = Division )  ) + 
  geom_line(aes(color = Division), linewidth = 1) + 
  geom_point(size = 3, shape = 21) +
  theme_bw() + 
  scale_x_discrete(limits =c("winter", "spring", "summer", "fall") )+
  scale_y_continuous( labels = function(x) paste0(x*100, "%")) + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(), 
        legend.background = element_blank(),
        legend.box.background = element_rect(colour = "gray"),
        legend.position = c(0.1, 0.8),
        axis.text.x = element_text(face = "bold", size = 12), 
        axis.text.y = element_text(face = "bold", size = 11), 
        axis.title = element_text(face = "bold", size = 14)) +
  scale_fill_manual(values = c("#1b9476", "#3492eb")) +
  scale_color_manual(values = c("#1b9476", "#3492eb")) +
  labs(x = "Season", y = "Relative abundance in season (%)") 
  
```


```{r}
#tax_table %>% 
# dplyr::mutate(Class = ifelse(Class == "", 
#                       paste0("Unclassified ", Division), Class)) -> tax_table

asv_table_r %>% 
  mutate_if(is.integer, as.numeric) %>% 
  mutate(class = tax_table$Class) %>%
  pivot_longer(cols = -class, 
               values_to = "abundance", 
               names_to = "sample") %>%
  left_join(.,env_lobau_final_r, by = c("sample" = "sample_id")) %>%  
  dplyr::select ("abundance", "class", "well_id.y", "season") %>%
  mutate(well_id.y = as.factor(well_id.y),
         class = as.factor(class),
         season = as.factor(season)) %>%  
# to find relative abundance within each category  
  dplyr::group_by(well_id.y, season) %>%
  mutate (rel_abund = abundance / sum(abundance)) %>%
  ungroup() %>%
# to sum up all the counts of one class from all the samples  
  dplyr::group_by(well_id.y, class, season) %>%
  summarise(rel_abund = sum(rel_abund)) %>%
  ungroup() %>%
  
  
  left_join(., tax_table[,c("Division", "Class" )] %>% 
              distinct(Class, .keep_all=TRUE) , 
            by = c("class" = "Class") ) %>% 
  as.data.frame() -> data1

# coloring
ColourPalleteMulti <- function(df, group, subgroup){

  # Find how many colour categories to create and the number of colours in each
  categories <- aggregate(as.formula(paste(subgroup, group, sep="~" )), df, function(x) length(unique(x)))
  category.start <- (scales::hue_pal(l = 100)(nrow(categories))) 
  # Set the top of the colour pallete
  category.end  <- (scales::hue_pal(l = 40)(nrow(categories))) 
  # set the bottom

  
  # Build Colour pallette
  colours <- unlist(lapply(1:nrow(categories),
                          function(i){
                            colorRampPalette(colors = c(category.start[i], category.end[i]))(categories[i,2])}))
  return(colours)
}

colours <-ColourPalleteMulti(data1, "Division", "class")

data1
# fix taxonomy for plotting
divisions <- c("Ciliophora", "Apusomonadidae",  "Conosa", "Ochrophyta",
               "Rhodophyta", "Opalozoa",  "Breviatea", "Centroheliozoa", 
               "Chlorophyta", "Choanoflagellida", "Apicomplexa", "Protalveolata_X",
               "Cryptophyta", "Dinoflagellata", "Lobosa", "Streptophyta", 
               "Cercozoa", "Discoba", "Eukaryota_XX", "Metamonada", 
               "Foraminifera", "Haptophyta", "Pseudofungi", "Mesomycetozoa",
               "Katablepharidophyta", "Sagenista",  "Perkinsea", "Hilomonadea",
               "Stramenopiles_X",     "Telonemia"    )

#........!!!!! this part of the code still needs adaption to recreate the microshades manually

df$Phylum[!df$Phylum %in% phylums] <- "Others"
df$Family[!df$Phylum %in% phylums] <- "Others"

"Other Protobacteria" <- df$Family[df$Phylum=="Proteobacteria" & 
 !df$Family %in% c('Alcaligenaceae','Enterobacteriaceae')] 

"Other Bacteroidetes" <- df$Family[df$Phylum=="Bacteroidetes" &
 !df$Family %in% c('Bacteroidaceae','Rikenellaceae','Porphyromonadaceae')] 

"Other Firmicutes" <- df$Family[df$Phylum=="Firmicutes" & 
 !df$Family %in% c('Lactobacillaceae','Clostridiaceae','Ruminococcaceae','Lachnospiraceae')] 
#........!!!!!



# plot
  ggplot(data1[order(data1$Division), ],  
         aes(x = well_id.y, y = rel_abund, 
             fill = class)) + 
  geom_bar(stat = "identity", aes(order = Division) ) +
  theme_bw() +
  scale_fill_manual("", values=colours) +
  #scale_fill_manual(values=randomcoloR::distinctColorPalette(27) ) +
  theme(panel.grid = element_blank(), legend.position="right",
        legend.text = element_text(size = 6),      # Adjust the text size
        legend.title = element_text(size = 6),
        legend.key.size = unit(0.4, "cm"),
        axis.title = element_text(size=14,face="bold"),
        axis.text.y = element_text(size=10,face="bold"),
        axis.text.x = element_text(angle = 90, face = "bold")) + 
  labs(x = "Sampling site",
       y = "Relative abundance %", 
       fill = "Class") +
  guides(fill=guide_legend(ncol = 1)) +
    scale_y_continuous(labels = scales::percent_format()) +
  facet_wrap(~season, ncol = 4)

  data1$Division %>% unique()
```



```{r}

asv_table_r %>% 
  mutate_if(is.integer, as.numeric) %>% 
  mutate(Division = tax_table$Division) %>%
  pivot_longer(cols = -Division, 
               values_to = "abundance", 
               names_to = "sample") %>%
  mutate(Division = as.factor(Division) ,
         sample = as.factor(sample))%>% 
  
  group_by(sample, Division) %>% 
  summarise(abundance = sum(abundance)) %>% 
  ungroup() %>% 
  
  left_join(.,env_lobau_final_r, by = c("sample" = "sample_id")) %>%  
  dplyr::select ("abundance", "Division", "well_id.y", "sample") %>% 
  mutate(well_id.y = as.factor(well_id.y)) %>%  
# to find relative abundance within each sampling site, but also within each sample  
  dplyr::group_by(sample) %>%
  mutate (rel_abund = abundance / sum(abundance)) %>%
  ungroup() %>%
  dplyr::filter(Division %in% c("Ochrophyta",
                                "Ciliophora",
                                "Dinoflagellata", 
                                "Cryptophyta",
                                "Conosa", 
                                "Cercozoa",
                                "Discoba")) %>% 
  mutate(Ecology  = case_when(Division %in% c( "Ciliophora", 
                                               "Cercozoa") ~ "consumer", 
                              Division %in% c("Discoba", "Dinoflagellata") ~  "myxothrop", 
                              Division %in% c("Conosa") ~  "parasite", 
                              Division %in% c("Ochrophyta", "Cryptophyta") ~ "phototroph")) %>% 
  dplyr::filter(rel_abund >0 ) %>% 

  ggplot(aes(y=reorder(Division, rel_abund), x=rel_abund)) + 
  geom_point(aes(color = Ecology, fill = Ecology), alpha =0.2, size = 3) +
  geom_point(aes(color = Ecology, fill = Ecology), alpha =0.2, size = 2) +
  geom_boxplot(aes(color = Ecology, fill = Ecology), alpha =0.3, outlier.colour = NA) +
  facet_wrap(~well_id.y, ncol = 1) + 
  theme_bw() +
  labs(x  = "Relative abundance %", y= NULL) +
  scale_fill_manual(values = c("blue", "orange2", "gold", "darkgreen")) + 
  scale_color_manual(values = c("blue", "orange2", "gold", "darkgreen")) +
  theme( legend.background = element_blank(),
        legend.box.background = element_rect(colour = "gray"),
        legend.position = c(0.8, 0.2), 
        panel.grid = element_blank(), 
        strip.text = element_text(size =10)) 

```

```{r}
asv_table_r %>% 
  mutate_if(is.integer, as.numeric) %>% 
  mutate(Division = tax_table$Division) %>%
  pivot_longer(cols = -Division, 
               values_to = "abundance", 
               names_to = "sample") %>%
  mutate(Division = as.factor(Division) ,
         sample = as.factor(sample))%>% 
  
  group_by(sample, Division) %>% 
  summarise(abundance = sum(abundance)) %>% 
  ungroup() %>% 
  
  left_join(.,env_lobau_final_r, by = c("sample" = "sample_id")) %>%  
  dplyr::select ("abundance", "Division", "well_id.y", "sample") %>% 
  mutate(well_id.y = as.factor(well_id.y)) %>%  
# to find relative abundance within each sampling site, but also within each sample  
  dplyr::group_by(sample) %>%
  mutate (rel_abund = abundance / sum(abundance)) %>%
  ungroup() %>%
  dplyr::filter(Division %in% c("Ochrophyta",
                                "Ciliophora",
                                "Dinoflagellata", 
                                "Cryptophyta",
                                "Conosa", 
                                "Cercozoa",
                                "Discoba")) %>% 
  
   dplyr::group_by(well_id.y, Division) %>%
  summarise(mean = mean(rel_abund) *100, sd = sd(rel_abund)*100) %>%
  ungroup() %>% 
  arrange(mean)
```









